src/Functions/FunctionHashID.h

```diff
@@ -0,0 +1,163 @@
+#pragma once
+
+#include <Common/config.h>
+
+#include <hashids.h>
+
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <DataTypes/DataTypeString.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/FunctionHelpers.h>
+#include <Functions/IFunction.h>
+
+#include <functional>
+#include <initializer_list>
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int BAD_ARGUMENTS;
+    extern const int ILLEGAL_COLUMN;
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int TOO_MANY_ARGUMENTS_FOR_FUNCTION;
+    extern const int TOO_FEW_ARGUMENTS_FOR_FUNCTION;
+}
+
+// hashid(string, salt)
+class FunctionHashID : public IFunction
+{
+public:
+    static constexpr auto name = "hashid";
+
+    static FunctionPtr create(ContextPtr) { return std::make_shared<FunctionHashID>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 0; }
+
+    bool isVariadic() const override { return true; }
+
+    bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return false; }
+
+    DataTypePtr getReturnTypeImpl(const ColumnsWithTypeAndName & arguments) const override
+    {
+        if (arguments.size() < 1)
+            throw Exception(ErrorCodes::TOO_FEW_ARGUMENTS_FOR_FUNCTION, "Function {} expects at least one argument", getName());
+
+        const auto & id_col = arguments[0];
+        if (!isUnsignedInteger(id_col.type))
+            throw Exception(
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT,
+                "First argument of function {} must be unsigned integer, got {}",
+                getName(),
+                arguments[0].type->getName());
+
+        if (arguments.size() > 1)
+        {
+            const auto & hash_col = arguments[1];
+            const auto * hash_col_type_const = typeid_cast<const ColumnConst *>(hash_col.column.get());
+            if (!isString(hash_col.type) || !hash_col_type_const)
```

Comment:
I believe we have a helper function isColumnConst
`if (!isString(hash_col.type) || !isColumnConst(*hash_col.column.get()))`